src/Simplex_tree/concept/FiltrationValue.h

```diff
@@ -5,19 +5,65 @@
  *    Copyright (C) 2014 Inria
  *
  *    Modification(s):
+ *      - 2024/08 Hannah Schreiber: Update of the concept after several additions to the Simplex tree.
  *      - YYYY/MM Author: Description of the modification
  */
 
 /** \brief Value type for a filtration function on a cell complex.
-  *
-  * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
-  * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq 
-  * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices 
-  * by increasing filtration values (breaking ties so as a simplex appears after 
-  * its subsimplices of same filtration value) provides an indexing scheme 
-  * (see IndexingTag).
-  */
-  struct FiltrationValue {
-    /** \brief Operator < is a StrictWeakOrdering. */
-    bool operator<(FiltrationValue f1, FiltrationValue f2);
-  };
+ *
+ * Needs to implement `std::numeric_limits<FiltrationValue>::has_infinity` and when it returns `true`,
+ * has to implement `std::numeric_limits<FiltrationValue>::infinity()`. If it returns `false`, has to
+ * implement `std::numeric_limits<FiltrationValue>::max()` instead.
+ *
+ * A <EM>filtration</EM> of a cell complex (see FilteredComplex) is
+ * a function \f$f:\mathbf{K} \rightarrow \mathbb{R}\f$ satisfying \f$f(\tau)\leq
+ * f(\sigma)\f$ whenever \f$\tau \subseteq \sigma\f$. Ordering the simplices
+ * by increasing filtration values (breaking ties so as a simplex appears after
+ * its subsimplices of same filtration value) provides an indexing scheme
+ * (see IndexingTag).
+ */
+struct FiltrationValue {
+  /**
+   * @brief Has to construct the default value of FiltrationValue. E.g., 0 for a numerical value or {} for a vector.
+   */
+  FiltrationValue(0);
+
+  // only for default ordering of filtration_vect_ in initialize_filtration
+  /**
+   * @brief Strictly smaller operator. If the filtration values are totally ordered, should be a StrictWeakOrdering.
+   */
+  bool operator<(FiltrationValue f1, FiltrationValue f2);
+  // only for prune_above_filtration
+  /**
+   * @brief Smaller or equal operator.
+   */
+  bool operator<=(FiltrationValue f1, FiltrationValue f2);
+  /**
+   * @brief Equality operator
+   */
+  bool operator==(FiltrationValue f1, FiltrationValue f2);
+  /**
+   * @brief Not equal operator
+   */
+  bool operator!=(FiltrationValue f1, FiltrationValue f2);
+
+  /**
+   * @brief Given two filtration values at which a simplex exists, returns the minimal union of births generating
+   * a lifetime including those two values. The overload for native arithmetic types like `double` or `int` is
+   * already implemented.
```

Comment:
... as the minimum?

Reply:
Not sure to understand the question? 
Perhaps, my description is not clear? I had trouble to explain this, as it is not really an union of points but an union of infinite bars represented by their birth.